dapps/marketplace/src/pages/transaction/_OfferAcceptedBuyer.js

```diff
@@ -0,0 +1,211 @@
+import React, { Component, Fragment } from 'react'
+import { fbt } from 'fbt-runtime'
+import dayjs from 'dayjs'
+
+import DisputeOffer from './mutations/DisputeOffer'
+import FinalizeOffer from './mutations/FinalizeOffer'
+
+import Stages from 'components/TransactionStages'
+import SendMessage from 'components/SendMessage'
+
+const CONFIRMED_OFFERS_KEY = 'confirmed_offers'
+class OfferAcceptedBuyer extends Component {
+  state = {
+    offerConfirmed: false
+  }
+
+  componentDidMount() {
+    const offersConfirmedString = localStorage.getItem(CONFIRMED_OFFERS_KEY)
+    const offer = this.props.offer
+
+    if (offersConfirmedString) {
+      this.setState({
+        offerConfirmed: JSON.parse(offersConfirmedString).includes(offer.id)
+      })
+    }
+
+    // if fractional for rent listing and the endDate has passed automatically set offer as confirmed
+    if (
+      offer.listing.category === 'schema.forRent' &&
+      offer.endDate &&
+      offer.listing.__typename === 'FractionalListing'
+    ) {
+      if (
+        dayjs(offer.endDate)
+          .add(1, 'day')
+          .isBefore(dayjs())
+      ) {
+        this.setState({ offerConfirmed: true })
+      }
+    }
+  }
+
+  confirmOffer(offerId) {
+    const offersConfirmed = JSON.parse(
+      localStorage.getItem(CONFIRMED_OFFERS_KEY) || '[]'
+    )
+    offersConfirmed.push(offerId)
+    this.setState({
+      offerConfirmed: true
+    })
+    localStorage.setItem(CONFIRMED_OFFERS_KEY, JSON.stringify(offersConfirmed))
```

Comment:
can we not use state for this?

Reply:
Leaving it solely to state will result in not surviving page refreshes. Ideally this would work cross device (persisted in the smart contract/ipfs). With the `localStorage` solution it works on a single device across page refreshes.

Reply:
Can we not get if the offer was confirmed from GraphQL?